views/user.go

```diff
@@ -0,0 +1,402 @@
+package views
+
+import (
+	"crypto/md5"
+	"encoding/hex"
+	"fmt"
+	"github.com/gorilla/mux"
+	"github.com/ystv/web-auth/public/templates"
+	"github.com/ystv/web-auth/user"
+	"log"
+	"math"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+)
+
+type (
+	// UsersTemplate represents the context for the user template
+	UsersTemplate struct {
+		Users                                 []UserStripped
+		UserID                                int
+		CurPage, NextPage, PrevPage, LastPage int
+		ActivePage                            string
+		Sort                                  struct {
+			Pages      int
+			Size       int
+			PageNumber int
+			Column     string
+			Direction  string
+			Search     string
+		}
+	}
+)
+
+// UsersFunc handles a users request
+func (v *Views) UsersFunc(w http.ResponseWriter, r *http.Request) {
+	session, _ := v.cookie.Get(r, v.conf.SessionCookieName)
+
+	c := v.getData(session)
+
+	var err error
+
+	if r.Method == "POST" {
+		err = r.ParseForm()
+		if err != nil {
+			err = v.errorHandle(w, err)
+			if err != nil {
+				log.Println(err)
+				return
+			}
+		}
+
+		column := r.FormValue("column")
+		direction := r.FormValue("direction")
+		search := r.FormValue("search")
+		var size int
+		sizeRaw := r.FormValue("size")
+		if sizeRaw == "all" {
+			size = 0
+		} else {
+			size, err = strconv.Atoi(sizeRaw)
+			if err != nil {
+				size = 0
+			} else if size <= 0 {
+				err = v.errorHandle(w, fmt.Errorf("invalid size, must be positive"))
+				if err != nil {
+					log.Println(err)
+					return
+				}
+			} else if size != 5 && size != 10 && size != 25 && size != 50 && size != 75 && size != 100 {
+				size = 25
+			}
+		}
+		valid := true
+		switch column {
+		case "userId":
+		case "name":
+		case "username":
+		case "email":
+		case "lastLogin":
+			break
+		default:
+			valid = false
+		}
+		switch direction {
+		case "asc":
+		case "desc":
+			break
+		default:
+			valid = false
+		}
+		r.Method = "GET"
```

Comment:
Why is it necessary to overwrite the method of the incoming request?

Reply:
If I don't overwrite it then it loops infinity on the post

Reply:
It's just the way it's been written

Reply:
That doesn't seem right - it looks like a lot of the code in this `if` is identical to the bit below it. This function may merit being broken apart a bit.

Reply:
The code is consolidated later, recently to writing this comment but it is consolidated eventually